view/propertyEditor.js

```diff
@@ -245,77 +244,118 @@ ACS.propertyEditor = function (modelList, // ACS.modelList
 				cell.appendChild(dropdownList);
 			}
 
-			//generate checkbox field for boolean
-			if (comboValues === '' && typetemp === ACS.dataType.BOOLEAN) {
+			if (isDynProp) {
+
+				log.debug('downloading (dynamic) properties list');
+				setBaseURI(ACS.areBaseURI + '/rest/');
+
 				cell = row[h + 1].insertCell(1);
-				boolInput = null;
-				boolInput = document.createElement("INPUT");
-				boolInput.setAttribute("type", "checkbox");
-				boolInput.setAttribute("value", valtemp);
-				if ((valtemp === "true") || (valtemp === "True")) {
-					boolInput.setAttribute("checked", true);
+				var id = propertyEdPanelCaptions.innerHTML;
+				var key = propName;
+				var currentValue = valtemp || "";
+				var elementId = h + "/1/" + valtemp;
+
+				fetchDynProperties(cell, elementId, currentValue, id, key);
+				function fetchDynProperties(cellToAdd, elementId, currentValue, id, key) {
+					getRuntimeComponentPropertyList(function(data, httpStatus) {
+						var entries = data ? JSON.parse(data) : null;
+						entries.unshift("");
```

Comment:
yes, makes sense, simpler just `[""]` instead of `new Array("")`

Reply:
no, sorry - better to move line `entries.unshift("");` in the else-statement in line `269`. @deinhofer's solution breaks the if-check on line `263`, since then `entries.length` will be `1`

Reply:
Like this https://github.com/asterics/WebACS/pull/36/commits/7feaf22054cf81ffe5e344b8e290cd61161ca996?

Reply:
No, in the 'else' branch entries could still be 'null' and hence entries.unshift would produce an error.

new suggestion do all the magic like this:
```javascript
var entries = data ? JSON.parse(data).unshift("") : [""];
```

Then you have an extra empty field in any case.
Then you have to change the if-condition to check if entries.length === 1

Reply:
simpler: 
`var entries = data ? JSON.parse(data).unshift("") : null;`
and then 
`if(!entries)`

it makes no sense to create an empty array just to check afterwards if the length == 1.

Reply:
But the if/else should make sure that entries is not null in the else branch, shouldn't it?

Reply:
ok, what I said is nonsense - then we would have the problem that if `JSON.parse(data)` is empty, that we do not come in the first if-statement... I should go home ^^

Reply:
just figured that what I said was nonsense too... :smile: I'll review later at leisure and push a new suggestion.

Reply:
What about following suggestion: https://github.com/asterics/WebACS/pull/36/commits/8f5d9c76a48891e049da135c1429df9d7a587eef ?

Reply:
Bingo!! :+1: loogs good!!

Reply:
> What about following suggestion: [8f5d9c7](https://github.com/asterics/WebACS/commit/8f5d9c76a48891e049da135c1429df9d7a587eef) ?

Bingo!! :+1: loogs good!!

Reply:
Still not ideal, but we'll refactor it when es6 features are available (https://github.com/asterics/WebACS/pull/36/commits/f9df73106e86ea6564be58f7f755a7c642efa2fd).

Note: It is not ensured that what we're looking at is actually an array (thus having the property `length`).

Reply:
The only way to make sure that `unshift()` is called correctly - at least the only I can think of - would be to use `instanceof`, wouldn't it? Like:

```js
  if (entries instanceof Array) {
     ...
  }
```
We could check if the property `length` is available too, but I think using `instanceof` is better in this case.

Reply:
In future changes, `getRuntimeComponentPropertyList()` should return a valid array or `null` otherwise. This way we could avoid dispersing this kind of logic and reduce the test to a simple `null` comparison for users of this functions. What do you think?

Reply:
Actually, the first suggestion is probably the best, and should work too... :grin:

```js
var entries = data ? JSON.parse(data) : [];
if (entries.length === 0) { 
   // input
} else {
   // selection
}
```

Reply:
> yes, makes sense, simpler just `[""]` instead of `new Array("")`

Would simply `[]` be possible too? Like:

```js
var entries = data ? JSON.parse(data) : [];
```

What syntax has broader support?

Reply:
both should be supported in all browsers, however I vote for `[]` since it is shorter, simpler and as it seems also faster: https://stackoverflow.com/questions/7375120/why-is-arr-faster-than-arr-new-array

Reply:
I like the shorter form too ... generally :smirk: